### DISCUSSION TITLE
Nested Child Physics Body

### DISCUSSION THREAD

**Discussion by wheatley19c** [enable3d/enable3d#146] [Help / Questions] [answered]
Hi all,

From the [FBX loader and animation](https://enable3d.io/examples/fbx-loader-and-animations.html) example, I am trying to turn the cube which is nested under the robot's left hand into a physics body for collision detection. Similar to how one would nest a melee weapon. 

I modified the code in the traverse function like so: 
` child.add(this.third.physics.add.box({ width: 2, height: 2, depth: 2 }))`
But the cube obviously drops to the floor and does not maintain its position under the left hand. 

I'm not sure how to proceed here, any help would be appreciated. Thanks!

**Comment by yandeu** [comment, marked as answer]
Just get a reference to the hand and update the box manually in the update loop:

```js
robot.traverse(child => {
  if (child.name === 'mixamorigLeftHandIndex1') {
    if (!this.box) {
      // collisionFlags 2 = kinematic
      // collisionFlags 6 = kinematic and ghost
      this.box = this.third.physics.add.box({ width: 1, height: 1, depth: 1, collisionFlags: 2 })
      this.box.customParams = { child }
    }
  }
})
```

```js
update() {
  if (this.box && this.box.body) {
    const child = this.box.customParams.child
    const pos = child.getWorldPosition(new THREE.Vector3())
    this.box.position.copy(pos)
    this.box.rotation.copy(child.rotation)
    this.box.body.needUpdate = true
  }
}
```

Just made a new example:
https://enable3d.io/examples/fbx-loader-and-animations-hitbox.html

**Reply by wheatley19c** [reply]
Brilliant, thank you.
